packages/zendesk-adapter/test/filters/custom_objects/custom_object_fields_order.test.ts

```diff
@@ -0,0 +1,168 @@
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import {
+  CORE_ANNOTATIONS,
+  ElemID, getChangeData,
+  InstanceElement,
+  ObjectType,
+  ReferenceExpression,
+  toChange,
+} from '@salto-io/adapter-api'
+import { elements as elementsUtils, filterUtils } from '@salto-io/adapter-components'
+import {
+  CUSTOM_OBJECT_FIELD_ORDER_TYPE_NAME,
+  CUSTOM_OBJECT_FIELD_TYPE_NAME,
+  CUSTOM_OBJECT_TYPE_NAME,
+  ZENDESK,
+} from '../../../src/constants'
+import filterCreator, {
+  customObjectFieldsOrderType, ORDER_FIELD,
+} from '../../../src/filters/custom_objects/custom_object_fields_order'
+import { createFilterCreatorParams } from '../../utils'
+import ZendeskClient from '../../../src/client/client'
+
+const { RECORDS_PATH } = elementsUtils
+type FilterType = filterUtils.FilterWith<'onFetch' | 'deploy'>
+
+const createCustomObjectField = (id: number, parent: InstanceElement): InstanceElement => new InstanceElement(
+  `customObjectField${id}`,
+  new ObjectType({ elemID: new ElemID(ZENDESK, CUSTOM_OBJECT_FIELD_TYPE_NAME) }),
+  { id },
+  undefined,
+  { [CORE_ANNOTATIONS.PARENT]: [new ReferenceExpression(parent.elemID, parent)] }
+)
+
+const createCustomObject = (id: number): InstanceElement => new InstanceElement(
+  `customObject${id}`,
+  new ObjectType({ elemID: new ElemID(ZENDESK, CUSTOM_OBJECT_TYPE_NAME) }),
+  { key: 'key' }
+)
+
+const createCustomObjectFieldsOrder = (name: string, fields: InstanceElement[], parent: InstanceElement)
+  : InstanceElement => new InstanceElement(
+  name,
+  customObjectFieldsOrderType,
+  { [ORDER_FIELD]: fields.map(field => new ReferenceExpression(field.elemID, field)) },
+  [ZENDESK, RECORDS_PATH, CUSTOM_OBJECT_FIELD_ORDER_TYPE_NAME, name],
+  { [CORE_ANNOTATIONS.PARENT]: [new ReferenceExpression(parent.elemID, parent)] }
+)
+
+describe('customObjectFieldsOrderFilter', () => {
+  let customObjectFieldsOrderFilter: FilterType
+  let putSpy: jest.SpyInstance
+  let customObject1: InstanceElement
+  let customObject2: InstanceElement
+  let customObjectField1: InstanceElement
+  let customObjectField2: InstanceElement
+  let customObjectField3: InstanceElement
+  beforeEach(() => {
+    jest.clearAllMocks()
+    const client = new ZendeskClient({
+      credentials: { username: 'a', password: 'b', subdomain: 'ignore' },
+    })
+    putSpy = jest.spyOn(client, 'put')
+    customObjectFieldsOrderFilter = filterCreator(createFilterCreatorParams({ client })) as FilterType
+    customObject1 = createCustomObject(1)
+    customObject2 = createCustomObject(2)
+    customObjectField1 = createCustomObjectField(1, customObject1)
+    customObjectField2 = createCustomObjectField(2, customObject1)
+    customObjectField3 = createCustomObjectField(3, customObject2)
+  })
+  describe('onFetch', () => {
+    it('should create order instance and order type', async () => {
```

Comment:
can you add more tests, for cases where id is not defined, or when there is not parent